pkg/daemon/controller/controller.go

```diff
@@ -202,6 +202,128 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 	return ctrl.Result{}, nil
 }
 
+// Cleanup removes an instance from the cluster. If the parameter flag is true, the node object is not deleted.
+func (sc *ServiceController) Cleanup(preserveNode bool) error {
```

Comment:
Why is `Cleanup` of services a function of the controller? Shouldn't the `Mgr` be the one to handle that?